lib/Sema/CSApply.cpp

```diff
@@ -5400,10 +5400,15 @@ Expr *ExprRewriter::coerceCallArguments(Expr *arg, AnyFunctionType *funcType,
 
       // Otherwise, create a call of the default argument generator.
       } else {
-        defArg =
-            new (tc.Context) DefaultArgumentExpr(callee, paramIdx,
-                                                 arg->getStartLoc(),
-                                                 param.getParameterType());
+        const ParamDecl *defaultParam = getParameterAt(cast<ValueDecl>(callee.getDecl()), newArgs.size());
+        defArg = defaultParam->getDefaultValue();
```

Comment:
This would still be broken if default is something like `= #line` or other magic literal or `nil`. I'll take a look at how happens with default arguments in SILGen for the normal calls, maybe that'll give us some insights how to be about this...

Reply:
@xedin it doesn't seem to be an issue. I tested with both `subscript(x: Int = #line)` and `subscript(x: Int? = nil)`. I'll update the tests.